app/src/main/java/ch/epfl/sdp/appart/login/LoginService.java

```diff
@@ -0,0 +1,52 @@
+package ch.epfl.sdp.appart.login;
+
+import ch.epfl.sdp.appart.user.User;
+
+/**
+ * Interface that any Login service should implement
+ */
+public interface LoginService {
+
+    /**
+     * Performs a user login with email and password.
+     *
+     * @param email    the user's email
+     * @param password the user's password
+     * @return a boolean that indicates if the login was successful
+     */
+    boolean loginWithEmail(String email, String password);
+
+    /**
+     * Performs a user login with username and password.
+     *
+     * @param username the user's username
+     * @param password the user's password
+     * @return a boolean that indicates if the login was successful
+     */
+    boolean loginWithUsername(String username, String password);
+
+    /**
+     * Retrieves the current user if there is one, null otherwise.
+     *
+     * @return the current user if logged-in, null otherwise
+     */
+    User getCurrentUser();
+
+    /**
+     * Resets the password associated to an email if said email is linked with an account
+     * We do not indicate if the email was found or not to prevent malicious users to know if someone is registered or not
+     *
+     * @param email the user's email
+     */
+    void resetPasswordWithEmail(String email);

```

Comment:
We could consider using bool as return type, not to indicate whether the email is linked to an account, but to indicate whether the query was successful (e.g. it could fail if there are issue with Firestore). For example, if True is returned then the view will show "If an account exists, you will receive an email shortly", in the other hand if False is returned the view will show "Sorry there was a problem, try again later".

Reply:
Ok, this makes sense.